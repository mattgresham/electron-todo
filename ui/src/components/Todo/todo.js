import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { ReactComponent as IconDelete } from '../../images/delete.svg';
import styles from './todo.module.css';

const Todo = ({ completeTodo, deleteTodo, id, isCompleted, text }) => (
  <div className={styles.container}>
    <div>
      <input
        checked={isCompleted}
        className={styles.completedCheckbox}
        disabled={isCompleted}
        id={`todo-checkbox-${id}`}
        onChange={() => completeTodo(id)}
        type="checkbox"
      />
      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */}
      <label className="sr-only" htmlFor={`todo-checkbox-${id}`}>
        Complete todo
      </label>
    </div>
    <span
      className={classnames(styles.text, {
        [styles.strikethrough]: isCompleted
      })}
    >
      {text}
    </span>
    <button
      className={styles.deleteButton}
      onClick={() => deleteTodo(id)}
      type="button"
    >
      <IconDelete title="Delete Todo" />
    </button>
  </div>
);

Todo.propTypes = {
  completeTodo: PropTypes.func.isRequired,
  deleteTodo: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  isCompleted: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired
};

export default Todo;
