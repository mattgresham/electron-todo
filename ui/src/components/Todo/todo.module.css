.container {
  display: flex;
  margin: 0.25rem 1rem 0.25rem 0;
}

.completedCheckbox {
  margin: 0.8rem 1rem 0.2rem 1rem;
}

.text {
  flex: 1;
  font-size: 2rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.strikethrough {
  text-decoration: line-through;
}

.deleteButton {
  background: none;
  border: 0;
  color: #cd5c5c;
  display: flex;
  padding-top: 0.25rem;
}

.deleteButton svg {
  height: 1.25rem;
  width: 1.25rem;
}
