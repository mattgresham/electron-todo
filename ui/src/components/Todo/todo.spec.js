import React from 'react';
import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Todo from '.';

configure({ adapter: new Adapter() });

it(`Todo renders with default props`, () => {
  const tree = render(
    <Todo
      completeTodo={() => {}}
      deleteTodo={() => {}}
      id={1}
      isCompleted={false}
      text="Todo"
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
});

it(`Todo renders with complete prop`, () => {
  const tree = render(
    <Todo
      completeTodo={() => {}}
      deleteTodo={() => {}}
      id={1}
      isCompleted
      text="Todo"
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
});
