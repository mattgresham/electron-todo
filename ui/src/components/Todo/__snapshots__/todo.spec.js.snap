// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders with complete prop 1`] = `
<div
  class="container"
>
  <div>
    <input
      checked=""
      class="completedCheckbox"
      disabled=""
      id="todo-checkbox-1"
      type="checkbox"
    />
    <label
      class="sr-only"
      for="todo-checkbox-1"
    >
      Complete todo
    </label>
  </div>
  <span
    class="text strikethrough"
  >
    Todo
  </span>
  <button
    class="deleteButton"
    type="button"
  >
    <svg
      title="Delete Todo"
    >
      delete.svg
    </svg>
  </button>
</div>
`;

exports[`Todo renders with default props 1`] = `
<div
  class="container"
>
  <div>
    <input
      class="completedCheckbox"
      id="todo-checkbox-1"
      type="checkbox"
    />
    <label
      class="sr-only"
      for="todo-checkbox-1"
    >
      Complete todo
    </label>
  </div>
  <span
    class="text"
  >
    Todo
  </span>
  <button
    class="deleteButton"
    type="button"
  >
    <svg
      title="Delete Todo"
    >
      delete.svg
    </svg>
  </button>
</div>
`;
